pinctrl: gecko: LEUART_TypeDef undefined.

**Describe the bug**
A twister build for a completly different pull-request for a gpio driver failes with latest mainline.
According to build-log the type `LEUART_TypeDef` is undefined.
I couldn't find the definition for this type in Zephyr at all.
I suppose this issue is linked to pull-request #60695

**To Reproduce**
Examples to reproduce:
* twister build for `tests/drivers/build_all/gpio` (see pull-reqest #59768)
* run `west build -p -b efr32xg24_dk2601b samples/basic/blinky` manually
* 


**Expected behavior**
`zephyr/drivers/pinctrl/pinctrl_gecko.c` should compile without error

**Impact**
Foremost my Pull-Request is blocked, but I assume that this will be an issue for other builds as well!

**Logs and console output**
```
[144/219] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_gecko.c.obj
FAILED: zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_gecko.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DEFR32BG22C224F512IM40 -DKERNEL -DSL_CATALOG_ZEPHYR_KERNEL_PRESENT -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/efr32bg22_brd4184a/tests/drivers/build_all/gpio/drivers.gpio.build/zephyr/include/generated -I/__w/zephyr/zephyr/soc/arm/silabs_exx32/efr32bg22 -I/__w/zephyr/zephyr/soc/arm/silabs_exx32/common/. -I/__w/zephyr/zephyr/boards/arm/efr32_thunderboard/. -I/__w/zephyr/zephyr/subsys/testsuite/include -I/__w/zephyr/modules/lib/acpica/../../zephyr/include -I/__w/zephyr/modules/lib/acpica/.. -I/__w/zephyr/modules/lib/acpica/source/include -I/__w/zephyr/modules/lib/acpica/source/include/platform -I/__w/zephyr/modules/lib/acpica/source/compiler -I/__w/zephyr/modules/lib/acpica/generate/zephyr -I/__w/zephyr/modules/lib/acpica/source/tools/acpiexec -I/__w/zephyr/modules/lib/acpica/source/tools/acpidump -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/hal/silabs/gecko/Device/SiliconLabs/EFR32BG22/Include -I/__w/zephyr/modules/hal/silabs/gecko/board/config/brd4184a -I/__w/zephyr/modules/hal/silabs/gecko/board/inc -I/__w/zephyr/modules/hal/silabs/gecko/common/inc -I/__w/zephyr/modules/hal/silabs/gecko/emlib/inc -I/__w/zephyr/modules/hal/silabs/gecko/platform/radio/rail_lib/chip/efr32/efr32xg2x -I/__w/zephyr/modules/hal/silabs/gecko/platform/radio/rail_lib/common -I/__w/zephyr/modules/hal/silabs/gecko/platform/radio/rail_lib/plugin/pa-conversions -I/__w/zephyr/modules/hal/silabs/gecko/platform/radio/rail_lib/plugin/pa-conversions/efr32xg22/config -I/__w/zephyr/modules/hal/silabs/gecko/protocol/bluetooth/bgstack/ll/inc -I/__w/zephyr/modules/hal/silabs/gecko/service/device_init/config/s2 -I/__w/zephyr/modules/hal/silabs/gecko/service/device_init/config/s2/sdid205 -I/__w/zephyr/modules/hal/silabs/gecko/service/device_init/inc -I/__w/zephyr/modules/hal/silabs/gecko/service/hfxo_manager/config -I/__w/zephyr/modules/hal/silabs/gecko/service/hfxo_manager/inc -I/__w/zephyr/modules/hal/silabs/gecko/service/hfxo_manager/src -I/__w/zephyr/modules/hal/silabs/gecko/service/power_manager/config -I/__w/zephyr/modules/hal/silabs/gecko/service/power_manager/inc -I/__w/zephyr/modules/hal/silabs/gecko/service/power_manager/src -I/__w/zephyr/modules/hal/silabs/gecko/service/sleeptimer/config -I/__w/zephyr/modules/hal/silabs/gecko/service/sleeptimer/inc -I/__w/zephyr/modules/hal/silabs/gecko/service/sleeptimer/src -I/__w/zephyr/modules/hal/silabs/gecko/util/third_party/crypto/sl_component/sl_protocol_crypto/src -I/__w/zephyr/zephyr/boards/arm/efr32_thunderboard -I/__w/zephyr/modules/hal/silabs/gecko/se_manager/src -I/__w/zephyr/modules/hal/silabs/gecko/se_manager/inc -isystem /__w/zephyr/zephyr/lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/include -isystem /opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/include-fixed -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/efr32bg22_brd4184a/tests/drivers/build_all/gpio/drivers.gpio.build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m33 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/tests/drivers/build_all/gpio=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_gecko.c.obj -MF zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_gecko.c.obj.d -o zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_gecko.c.obj -c /__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c
/__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c: In function 'pinctrl_configure_pins':
/__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c:14:9: error: unknown type name 'LEUART_TypeDef'; did you mean 'EUSART_TypeDef'?
   14 |         LEUART_TypeDef *lebase = (LEUART_TypeDef *)reg;
      |         ^~~~~~~~~~~~~~
      |         EUSART_TypeDef
/__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c:14:35: error: 'LEUART_TypeDef' undeclared (first use in this function); did you mean 'EUSART_TypeDef'?
   14 |         LEUART_TypeDef *lebase = (LEUART_TypeDef *)reg;
      |                                   ^~~~~~~~~~~~~~
      |                                   EUSART_TypeDef
/__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c:14:35: note: each undeclared identifier is reported only once for each function it appears in
/__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c:14:51: error: expected expression before ')' token
   14 |         LEUART_TypeDef *lebase = (LEUART_TypeDef *)reg;
      |                                                   ^
/__w/zephyr/zephyr/drivers/pinctrl/pinctrl_gecko.c:14:25: error: unused variable 'lebase' [-Werror=unused-variable]
   14 |         LEUART_TypeDef *lebase = (LEUART_TypeDef *)reg;
      |                         ^~~~~~
cc1: all warnings being treated as errors
```

**Environment (please complete the following information):**
- Github twister pipeline

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

--- a/drivers/pinctrl/pinctrl_gecko.c
+++ b/drivers/pinctrl/pinctrl_gecko.c
@@ -11,11 +11,17 @@
 int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintptr_t reg)
 {
 	USART_TypeDef *base = (USART_TypeDef *)reg;
-	LEUART_TypeDef *lebase = (LEUART_TypeDef *)reg;
 	uint8_t loc;
-#ifndef CONFIG_SOC_GECKO_SERIES1
+#ifdef CONFIG_SOC_GECKO_SERIES1
+	LEUART_TypeDef *lebase = (LEUART_TypeDef *)reg;
+#else
 	int usart_num = USART_NUM(base);
 #endif
+
+#ifdef CONFIG_UART_GECKO
+	struct soc_gpio_pin rxpin = {0, 0, 0, 0};
+	struct soc_gpio_pin txpin = {0, 0, 0, 0};
+#endif /* CONFIG_UART_GECKO */
 
 	struct soc_gpio_pin pin_config = {0, 0, 0, 0};
 
@@ -27,17 +33,21 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintp
 		switch (GECKO_GET_FUN(pins[i])) {
 #ifdef CONFIG_UART_GECKO
 		case GECKO_FUN_UART_RX:
-			pin_config.mode = gpioModeInput;
-			pin_config.out = 1;
-			GPIO_PinModeSet(pin_config.port, pin_config.pin, pin_config.mode,
-					pin_config.out);
+			rxpin.port = GECKO_GET_PORT(pins[i]);
+			rxpin.pin = GECKO_GET_PIN(pins[i]);
+			rxpin.mode = gpioModeInput;
+			rxpin.out = 1;
+			GPIO_PinModeSet(rxpin.port, rxpin.pin, rxpin.mode,
+					rxpin.out);
 			break;
 
 		case GECKO_FUN_UART_TX:
-			pin_config.mode = gpioModePushPull;
-			pin_config.out = 1;
-			GPIO_PinModeSet(pin_config.port, pin_config.pin, pin_config.mode,
-					pin_config.out);
+			txpin.port = GECKO_GET_PORT(pins[i]);
+			txpin.pin = GECKO_GET_PIN(pins[i]);
+			txpin.mode = gpioModePushPull;
+			txpin.out = 1;
+			GPIO_PinModeSet(txpin.port, txpin.pin, txpin.mode,
+					txpin.out);
 			break;
 
 #ifdef CONFIG_SOC_GECKO_SERIES1
@@ -105,11 +115,11 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintp
 			GPIO->USARTROUTE[usart_num].ROUTEEN =
 				GPIO_USART_ROUTEEN_TXPEN | GPIO_USART_ROUTEEN_RXPEN;
 			GPIO->USARTROUTE[usart_num].TXROUTE =
-				(pin_config.pin << _GPIO_USART_TXROUTE_PIN_SHIFT) |
-				(pin_config.port << _GPIO_USART_TXROUTE_PORT_SHIFT);
+				(txpin.pin << _GPIO_USART_TXROUTE_PIN_SHIFT) |
+				(txpin.port << _GPIO_USART_TXROUTE_PORT_SHIFT);
 			GPIO->USARTROUTE[usart_num].RXROUTE =
-				(pin_config.pin << _GPIO_USART_RXROUTE_PIN_SHIFT) |
-				(pin_config.port << _GPIO_USART_RXROUTE_PORT_SHIFT);
+				(rxpin.pin << _GPIO_USART_RXROUTE_PIN_SHIFT) |
+				(rxpin.port << _GPIO_USART_RXROUTE_PORT_SHIFT);
 #endif /* CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION */
 
 #ifdef UART_GECKO_HW_FLOW_CONTROL

